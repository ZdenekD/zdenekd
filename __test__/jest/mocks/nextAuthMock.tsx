import {Session} from 'next-auth';
import {useSession} from 'next-auth/react';

jest.mock('next-auth/react');

const mockSession: Session = {
    expires: '1',
    user: {
        name: 'Name',
        email: 'email@email.com ',
        image: '',
    },
};

(useSession as jest.Mock).mockReturnValueOnce([mockSession, false]);
