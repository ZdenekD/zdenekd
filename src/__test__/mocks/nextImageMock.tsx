const nextImageMock = jest.mock('next/image', () => function Image() {
    return <img src="" alt="" />;
});

export default nextImageMock;
