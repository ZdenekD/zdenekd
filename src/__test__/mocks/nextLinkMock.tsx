const nextLinkMock = jest.mock('next/link', () => function Link() {
    return <a href="/">link</a>;
});

export default nextLinkMock;
