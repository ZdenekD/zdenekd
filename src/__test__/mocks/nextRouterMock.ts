const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'en',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
