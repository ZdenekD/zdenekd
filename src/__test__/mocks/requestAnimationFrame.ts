import react from 'react';

window.requestAnimationFrame = () => 0;

export default react;
