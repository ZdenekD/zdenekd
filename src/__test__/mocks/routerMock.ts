const routerMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
    })),
}));

export default routerMock;
