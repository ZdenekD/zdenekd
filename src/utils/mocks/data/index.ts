import MockDataParagraph from './content/UI/paragraph';

export {MockDataParagraph};
