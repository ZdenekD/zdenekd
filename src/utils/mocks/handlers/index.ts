import handler from './handler';

const handlers = [...handler];

export default handlers;
