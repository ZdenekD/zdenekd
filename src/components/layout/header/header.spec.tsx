import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '../../../__test__/utils/helpers';
import '../../../__test__/mocks/routerMock';
import Header from '.';

describe('Header', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<Header />);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-header');

        expect(component.exists()).toBe(true);
    });
});
