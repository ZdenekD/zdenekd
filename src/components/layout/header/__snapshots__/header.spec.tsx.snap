// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header match snapshot 1`] = `
<header class="header"
        data-test="component-header"
>
  <a href="/"
     class="link "
     aria-label="Odkaz na domovskou stránku"
  >
    <div class="wrapper ">
      <svg data-filename="src/assets/images/logo.svg">
      </svg>
      <i class="shadow"
         style="opacity:1"
      >
      </i>
    </div>
  </a>
  <button type="button"
          class="button closed "
          data-test="component-menu"
          aria-label="Otevřít navigaci"
  >
    <svg class="icon"
         viewbox="0 0 100 100"
    >
      <path d="M30,40 L70,40 Z">
      </path>
      <path d="M30,50 L70,50 Z">
      </path>
      <path d="M30,60 L70,60 Z">
      </path>
    </svg>
  </button>
</header>
`;
